File: test/data/cs/test03/StateConfiguring.cs
public class StateConfiguring : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateConfiguring(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitEvConfig()
    {
        _controllee.DoAction2();
        _stateController.InstanceOfIdle.Setup();
        return _stateController.InstanceOfIdle;
    }
    public override string GetStateName()
    {
        return "NotShooting.Configuring";
    }
    public override BaseState? GetParent()
    {
        return _stateController.InstanceOfNotShooting;
    }
}

File: test/data/cpp/test03/StateConfiguring.cpp
#include "StateConfiguring.h"
StateConfiguring::StateConfiguring(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateConfiguring::~StateConfiguring()
{
}
BaseState* StateConfiguring::TransitEvConfig()
{
    _controllee->DoAction2();
    _stateController->InstanceOfIdle->Setup(false, false);
    return _stateController->InstanceOfIdle;
}
const char* StateConfiguring::GetStateName()
{
    return "NotShooting.Configuring";
}
BaseState* StateConfiguring::GetParent()
{
    return _stateController->InstanceOfNotShooting;
}
int StateConfiguring::GetStateID()
{
    return 0;
}